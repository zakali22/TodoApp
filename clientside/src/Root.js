import React from "react";
import ReactDOM from "react-dom";
import "./index.scss";
import App from "./components/App";

import { createStore, applyMiddleware, compose } from "redux";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import thunk from "redux-thunk";
import rootReducer from "./reducers";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";

const initialState = {};
const middleware = [thunk];

const persistConfig = {
  key: "root",
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

let store = createStore(
  persistedReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
);

let persistor = persistStore(store);

export default props => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        {props.children}
      </PersistGate>
    </Provider>
  );
};
