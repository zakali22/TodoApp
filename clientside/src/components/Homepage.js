import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

class Homepage extends Component {
  render() {
    return (
      <div className="homepage">
        <div className="container">
          <div className="homepage__welcome">
            <h1>
              Organize your tasks with simple <span>todos</span>
            </h1>
            <p>
              No more fancy and hard to navigate UI's. <span>Just todo it</span>
              .
            </p>
          </div>
          <div className="homepage__register">
            <Link to="/register">
              <button>Get started</button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Homepage;
