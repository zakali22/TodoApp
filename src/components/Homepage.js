import React, { Component } from "react";
import PropTypes from "prop-types";

class Homepage extends Component {
  render() {
    return (
      <div className="homepage">
        <div className="homepage__welcome">
          <h1>Organize your tasks with simple todos</h1>
          <p>No more fancy and hard to navigate UI's</p>
        </div>
        <div className="homepage__register">
          <button>Get started</button>
        </div>
      </div>
    );
  }
}

export default Homepage;
