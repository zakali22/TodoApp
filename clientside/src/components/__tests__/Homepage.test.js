import React from "react";
import ReactDOM from "react-dom";
import { mount, shallow } from "enzyme";
import Homepage from "../Homepage";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Root from "../../Root";
import sinon from "sinon";

describe("Homepage", () => {
  test("has a valid snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Root>
          <Homepage />
        </Root>
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe("componentDidMount", () => {
    let push, props, wrapper;

    beforeEach(() => {
      props = {
        history: {
          push: jest.fn()
        },
        auth: true
      };
      push = sinon.stub(props.history, "push");
    });
    afterEach(() => {
      push.reset();
    });
    it("redirects if auth is true", () => {
      wrapper = mount(
        <MemoryRouter>
          <Root>
            <Homepage {...props} />
          </Root>
        </MemoryRouter>
      );
      wrapper.update();
      expect(push.calledOnce).toBe(true);
    });
  });
});
