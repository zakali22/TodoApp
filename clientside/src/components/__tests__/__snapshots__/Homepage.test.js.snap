// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage has a valid snapshot 1`] = `
<div
  className="homepage"
>
  <div
    className="container"
  >
    <div
      className="homepage__welcome"
    >
      <h1>
        Organize your tasks with simple 
        <span>
          todos
        </span>
      </h1>
      <p>
        No more fancy and hard to navigate UI's. 
        <span>
          Just todo it
        </span>
        .
      </p>
    </div>
    <div
      className="homepage__register"
    >
      <a
        href="/register"
        onClick={[Function]}
      >
        <button>
          Get started
        </button>
      </a>
    </div>
  </div>
</div>
`;
